homeassistant/components/threshold/binary_sensor.py

```diff
@@ -260,6 +283,14 @@ def above(sensor_value: float, threshold: float) -> bool:
             self._state = None
             return
 
+        # guard against the case where the thresholds are not set
+        if not hasattr(self, "_threshold_lower") and not hasattr(
+            self, "_threshold_upper"
+        ):
+            self._state_position = POSITION_UNKNOWN
+            self._state = None
+            return
```

Comment:
Why is this case added? If it's only needed for preview, it should be possible to handle it in `async_start_preview`, if it's a bug fix it should be moved to a separate PR.

Reply:
Yea, that's just the case for preview. I moved this to `async_start_preview`. See the new PR as I screwed up the rebase